torch/_inductor/fx_passes/quantization.py

```diff
@@ -1364,11 +1392,66 @@ def _generate_qlinear_weight_prepack_patterns(dtype=torch.float32):
     )
 
 
-@functools.lru_cache(None)
-def _register_quantization_weight_pack_pass():
-    for dtype in [torch.float32, torch.bfloat16]:
-        # Step 1: Dequant promotion for int8-mixed-fp32/bf16
-        # Transform
+def _register_dequant_promotion():
+    dequant_pattern_cases = itertools.product(
+        [torch.float32, torch.bfloat16], [True, False]
+    )
+    for dtype, input_dim_exceeds_two in dequant_pattern_cases:
+        # 4 dequantization patterns will be matched based on the dtype and input dimension size.
```

Comment:
Instead of enumerating all the 4 patterns in the comment, maybe we can simplify it with a single diagram with optional ops (reshape and to_bf16) marked in it?

Reply:
Thanks for the suggestion. Simplified it into a single diagram with optional ops.